tutor/commands/k8s.py

```diff
@@ -49,6 +50,43 @@ def core_api(self):  # type: ignore
             self._core_api = self._client.CoreV1Api()
         return self._core_api
 
+    def is_running_inside_pod(self) -> bool:
+        # pylint: disable=import-outside-toplevel
+        from kubernetes.config.incluster_config import (
+            SERVICE_HOST_ENV_NAME,
+            SERVICE_PORT_ENV_NAME,
+        )
+
+        return (
+            SERVICE_HOST_ENV_NAME in os.environ and SERVICE_PORT_ENV_NAME in os.environ
+        )
+
+    def kube_config_exists(self) -> bool:
+        # pylint: disable=import-outside-toplevel
+        from kubernetes import config
+
+        kube_config_path = config.kube_config.KUBE_CONFIG_DEFAULT_LOCATION
+        return os.path.exists(kube_config_path)
+
+    def load_config(self) -> None:
```

Comment:
Your implementation is top-notch. I just have two minor comments:

1. Would you like to simplify the code by removing the `kube_config_exists` and `is_running_inside_pod` functions, and inlining the corresponding call? The functions are not used anywhere else, and they are quite simple, so I figure we can just remove them. I don't have a strong opinion, so follow your judgment here.
2. Exception management: I think that if either `load_kube_config` or `load_incluster_config` calls fail, then we should just let Tutor crash, so there is no need to catch the ConfigException. On the other hand, if we detect early that there is no kube config file or cluster env vars, then we should raise a TutorException that will be caught by the Tutor main process. In other words:

```
if self.kube_config_exists():
    # Loading kubeconfig file.
    config.load_kube_config()
elif self.is_running_inside_pod():
    # Loading in-cluster configuration.
    config.load_incluster_config()
else:
    raise TutorError(f"No Kubernetes configuration file in {config.kube_config.KUBE_CONFIG_DEFAULT_LOCATION} and service environment variables {SERVICE_HOST_ENV_NAME}, {SERVICE_PORT_ENV_NAME} are not defined. Configure your host to connect to blablabla (add intelligent suggestions here)")
```